### PR TITLE
feature/se-46 Employee Creation Page

### LINKED ISSUE #46: Create the Front-end for Employee Creation
image

### PR DESCRIPTION
**Pull request by Arkcorner**
85zr2q-1

closes https://github.com/SE2-WS23/documentation/issues/46

The Material UI Library sadly does not include an upload box to drag and drop files into currently as they are currently being developed. So this was made with a MUI button that lets you select the Files to upload

### CODE REVIEW

**Comment by julix14** [comment]
@Arkcorner please resolve any merge conflicts before marking it ready to review

**Comment by Arkcorner** [comment]
> @Arkcorner please resolve any merge conflicts before marking it ready to review
merge conflict resolved

**Comment by Arkcorner** [comment]
Accident

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+ * @returns {JSX.Element} The Employee creation page component.
+ */
+
+function EmployeeCreationPage() {
+
+  const [employeeName, setEmployeeName] = useState("");
+  const [employeeEmail, setEmployeeEmail] = useState("");
+  const [employeePhoneNumber, setEmployeePhoneNumber] = useState("");
+  const [employeeAddress, setEmployeeAddress] = useState("");
+  const [employeeExtraInfo, setEmployeeExtraInfo] = useState("");
```
Comment (Ara-Ara-ASid-kun) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx:18]: Hi, { useState } is not imported, pls import, UwU 

image

**Review by Ara-Ara-ASid-kun — CHANGES_REQUESTED** [review]

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx (hunk 1 above)
Comment (Arkcorner) [reply src/pages/EmployeeCreation/EmployeeCreationPage.jsx:18]: done UwU

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import PageLayout from "../../components/PageLayout";
+import { useState } from "react";
+
+/**
+ * The `EmployeeCreationPage` function is a React component that renders a form for users to create
+ * a new employee, including fields for the employee's name, email, phone number, address, and
+ * additional information, as well as a button to upload a resume and a button to create the employee.
+ * @returns {JSX.Element} The Employee creation page component.
+ */
+
```
Comment (julix14) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx]:
```suggestion
```

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    setEmployeePhoneNumber(event.target.value);
+  }
+  const handleEmployeeAddressChange = (event) => {
+    setEmployeeAddress(event.target.value);
+  }
+  const handleEmployeeExtraInfoChange = (event) => {
+    setEmployeeExtraInfo(event.target.value);
+  }
+
+  function EmployeeCreation(event) {
```
Comment (julix14) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx]:
```suggestion
  function handleSubmit(event) {
```

Function names should be camelCase

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+      <Typography variant="h6">Submit Files </Typography>
+      <UploadBox title="Resume" />
+        </Box>
+
+        <Button 
+        type="submit"
+        variant="contained"
+        color="primary"
+        size="large"
+        onClick={EmployeeCreation}>Create Employee
```
Comment (julix14) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx]:
```suggestion
        onClick={handleSubmit}>Create Employee
```

File: src/pages/EmployeeCreation/components/UploadBox.jsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import React, { useState } from "react";
+import { Button, Typography, Box } from "@mui/material";
+import UploadedFile from "./UploadedFile";
+
+
+/**
+ * @returns {JSX.Element} The Box holding all uploaded files in the Employee creation page component.
+ */
+
```
Comment (julix14) [inline src/pages/EmployeeCreation/components/UploadBox.jsx]:
```suggestion
```

File: src/pages/EmployeeCreation/components/UploadedFile.jsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import { Typography, Box, IconButton } from "@mui/material";
+import { InsertDriveFile } from "@mui/icons-material";
+import DeleteIcon from '@mui/icons-material/Delete';
+import PropTypes from 'prop-types'
+
+
+/**
+ * @returns {JSX.Element} The File Component for Employee creation page to display uploaded Files.
+ */
+
```
Comment (julix14) [inline src/pages/EmployeeCreation/components/UploadedFile.jsx]:
```suggestion
```

File: src/pages/EmployeeCreation/components/UploadBox.jsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+      {selectedFiles.map((file) => (
+    <Box key={file.name} display="flex" alignItems="center" mb={1}>
+      <UploadedFile />
+      <Typography variant="subtitle1" style={{ marginLeft: 8 }}>
+      {file.name}
+      </Typography>
+    </Box>
+    ))}
+      <label htmlFor="upload-file">
+        <input
```
Comment (julix14) [inline src/pages/EmployeeCreation/components/UploadBox.jsx]: https://mui.com/material-ui/api/input/
Use Mui components

File: src/pages/EmployeeCreation/components/UploadBox.jsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    const newFiles = Array.from(files);
+    setSelectedFiles((prevFiles) => [...prevFiles, ...newFiles]);
+  };
+
+  const handleFileUpload = (event) => {
+    document.getElementById("upload-file").click();
+  }
+
+  return (
+    <div>
```
Comment (julix14) [inline src/pages/EmployeeCreation/components/UploadBox.jsx]: https://mui.com/material-ui/api/box/

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+    
+    flexDirection="column">
+      <TextField
+        margin="normal"
+        type="submit"
+        label="Extra info" 
+        minRows={6}
+        sx={{ minWidth:"40vw"}}
+        value={employeeExtraInfo}
+        multiline
```
Comment (julix14) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx]: This doesn't look correctly formatted

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+  </Box>
+      <Box >
+      <TextField  
+      label="Name"  margin="normal" sx={{ minWidth:"20vw"}} value={employeeName} onChange={handleEmployeeNameChange}/>
+      <TextField  
+      label="Email" margin="normal" sx={{ minWidth:"20vw"}} value={employeeEmail} onChange={handleEmployeeEmailChange}/>
+      
+      </Box>
+      <Box display="flex" flexDirection="row">
+      <TextField  label="Phone number" margin="normal" sx={{ minWidth:"20vw"}} value={employeePhoneNumber} onChange={handleEmployeePhoneNumberChange}/>
```
Comment (julix14) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx]: Formatting...

**Review by julix14 — CHANGES_REQUESTED** [review]
Format all files and use MUI Components when possible

File: src/pages/EmployeeCreation/components/UploadBox.jsx (hunk 7 above)
Comment (Arkcorner) [reply src/pages/EmployeeCreation/components/UploadBox.jsx]: done

File: src/pages/EmployeeCreation/components/UploadBox.jsx (hunk 8 above)
Comment (Arkcorner) [reply src/pages/EmployeeCreation/components/UploadBox.jsx]: done

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx (hunk 9 above)
Comment (Arkcorner) [reply src/pages/EmployeeCreation/EmployeeCreationPage.jsx]: now same formatting as employee login

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx (hunk 10 above)
Comment (Arkcorner) [reply src/pages/EmployeeCreation/EmployeeCreationPage.jsx]: done

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+      <FormGroup className="employeeCreationForm">
+        <Box
+          sx={{
+            display: "flex",
+            justifyContent: "center",
+            alignItems: "center",
+            minHeight: "100vh",
+            minWidth: "100vw",
+            flexDirection: "column"
+          }}
```
Comment (julix14) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx:61]:
```suggestion
        <Box
        display="flex"
        flexDirection="column"
        alignItems="center"
          sx={{
            minHeight: "100vh",
            minWidth: "100vw",
          }}
```

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+        >
+          <Box maxWidth="50%">
+            <Box display="flex" flexDirection="row" alignItems="center" justifyContent="center">
```
Comment (julix14) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx]:
```suggestion
            <Box>
```

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+                onClick={changeAvatar}
+              >
+                +
+              </Fab>
+            </Box>
+            <Box>
+              <TextField label="Name" margin="normal" sx={{ minWidth: "20vw" }} value={employeeName} onChange={handleEmployeeNameChange} />
+              <TextField label="Email" margin="normal" sx={{ minWidth: "20vw" }} value={employeeEmail} onChange={handleEmployeeEmailChange} />
+            </Box>
+            <Box display="flex" flexDirection="row">
```
Comment (julix14) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx]:
```suggestion
            <Box>
```

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+              <TextField label="Phone number" margin="normal" sx={{ minWidth: "20vw" }} value={employeePhoneNumber} onChange={handleEmployeePhoneNumberChange} />
+              <TextField label="Address" margin="normal" sx={{ minWidth: "20vw" }} value={employeeAddress} onChange={handleEmployeeAddressChange} />
+            </Box>
+            <Box display="flex" flexDirection="column">
```
Comment (julix14) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx]:
```suggestion
            <Box sx={{ flexDirection:"column"}}>
```

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+            justifyContent: "center",
+            alignItems: "center",
+            minHeight: "100vh",
+            minWidth: "100vw",
+            flexDirection: "column"
+          }}
+        >
+          <Box maxWidth="50%">
+            <Box display="flex" flexDirection="row" alignItems="center" justifyContent="center">
+              <Avatar sx={{ width: "10vw", height: "20vh", variant: "circular" }} />
```
Comment (julix14) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx:65]: image

Comment (julix14) [reply src/pages/EmployeeCreation/EmployeeCreationPage.jsx:65]: It should stay rounded

File: src/pages/EmployeeCreation/components/UploadedFile.jsx [hunk 16]
```diff
@@ -0,0 +1,35 @@
+import React from "react";
```
Comment (julix14) [inline src/pages/EmployeeCreation/components/UploadedFile.jsx:1]: image

If you upload a file, the following error occurs

File: src/pages/EmployeeCreation/components/UploadBox.jsx [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+ */
+function UploadBox() {
+  const [selectedFiles, setSelectedFiles] = useState([]);
+
+  const handleFileSubmit = (event) => {
+    console.log("File has been deleted");
+    const files = event.target.files;
+    const newFiles = Array.from(files);
+    setSelectedFiles((prevFiles) => [...prevFiles, ...newFiles]);
+  };
```
Comment (julix14) [inline src/pages/EmployeeCreation/components/UploadBox.jsx:17]: This is not working

Comment (julix14) [reply src/pages/EmployeeCreation/components/UploadBox.jsx:17]: It just wrote in the console

**Review by julix14 — CHANGES_REQUESTED** [review]

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx (hunk 10 above)
Comment (julix14) [reply src/pages/EmployeeCreation/EmployeeCreationPage.jsx]: The formatting is still not done properly, have you used prettier?

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx (hunk 15 above)
Comment (Chrissydrx) [reply src/pages/EmployeeCreation/EmployeeCreationPage.jsx:65]: Yeah, that would be important

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx [hunk 18]
```diff
@@ -0,0 +1,121 @@
...
+  function handleSubmit(event) {
+    console.log("Employee has been created");
+    console.log("Name of employee: " + employeeName);
+    console.log("Email of employee: " + employeeEmail);
+    console.log("Phone number of employee: " + employeePhoneNumber);
+    console.log("Address of employee: " + employeeAddress);
+  }
+  function changeAvatar(event) {
+    console.log("Avatar has been changed");
+  }
```
Comment (Chrissydrx) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx:45]: Why is this not called handleAvatarChange?

File: src/pages/EmployeeCreation/components/UploadBox.jsx [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+
+
+/**
+ * @returns {JSX.Element} The Box holding all uploaded files in the Employee creation page component.
+ */
+function UploadBox() {
+  const [selectedFiles, setSelectedFiles] = useState([]);
+
+  const handleFileSubmit = (event) => {
+    console.log("File has been deleted");
```
Comment (Chrissydrx) [inline src/pages/EmployeeCreation/components/UploadBox.jsx:13]: File will not be deleted from the list

File: src/pages/EmployeeCreation/EmployeeCreationPage.jsx
Comment (Chrissydrx) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx:1]: Validation for inputs (emails, phone) would be nice

Comment (Chrissydrx) [inline src/pages/EmployeeCreation/EmployeeCreationPage.jsx:1]: That button looks weird

**Review by Chrissydrx — CHANGES_REQUESTED** [review]

**Comment by Chrissydrx** [comment]
More than 30 Changes, please fix everything and resolve merge conflicts and open a new PR

**Pull request closed without merging**